sys/uri_parser/uri_parser.c

```diff
@@ -133,6 +133,25 @@ static char *_consume_authority(uri_parser_result_t *result, char *uri,
         if (ipv6_end >= authority_end) {
             return NULL;
         }
+
+        char *zoneid_start = _strchrb(result->host, uri_end, '%');
```

Comment:
I think that something is wrong here around the length passed to strchrb.

The test I added to run into this was

```patch
+    VEC("coap://[fe80::1]/foo%20bar",
+        true,
+        "coap",
+        "",
+        "[fe80::1]",
+        "fe80::1",
+        "",
+        "",
+        "/foo%20bar",
+        "",
+        0),
```

that, on debug, showed

```
$2 = {scheme = 0x5655ec20 <validate_uris+7168> "coap://[fe80::1]/foo%20bar", userinfo = 0x0, 
  host = 0x5655ec27 <validate_uris+7175> "[fe80::1]/foo%20bar", ipv6addr = 0x5655ec28 <validate_uris+7176> "fe80::1]/foo%20bar", 
  zoneid = 0x5655ec35 <validate_uris+7189> "20bar", port = 0x0, path = 0x5655ec30 <validate_uris+7184> "/foo%20bar", query = 0x0, 
  scheme_len = 4, userinfo_len = 0, host_len = 9, ipv6addr_len = 12, zoneid_len = 65530, port_len = 0, path_len = 10, query_len = 0}
```

(Note how ipv6addr_len is longer than host_len).

Reply:
>char *zoneid_start = _strchrb(result->host, uri_end, '%');

did a fixup: replaced `uri_end` with `ipv6_end`. I also added your test vector to the unittests.